src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,277 @@
+	private JPanel createControlPanel() {
+		controlPanel = new JPanel();
+		controlPanel.setLayout(new GridLayout(1, 3));
+		
+		
+		JButton clearButton = new JButton("Clear Metrics");
+		JButton deleteButton = new JButton("Delete Rule");
+		JButton saveButton = new JButton("Save Rule");
+		
+		
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				System.out.println("Clear button pressed!");
+				clearMetricsListPanel();
+			}
+		});
+		
+		deleteButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					System.out.println("Delete button pressed!");
+					System.out.println("Rule name: " + nameText.getText());
+				}
+			}
+		});
+		
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					System.out.println("Save button pressed!");
+					System.out.println("Rule name: " + nameText.getText());
+				}
+			}
+		});
+		
+		controlPanel.add(clearButton);
+		controlPanel.add(deleteButton);
+		controlPanel.add(saveButton);
+		return controlPanel;
+
+	}
+	
+	/**
+	 * This method both returns and creates the JPanel holding the line which allows
+	 * users to add new metrics into the metrics list. TODO: The currently
+	 * hard-coded values of the JBoxes are to be replaced with the correct, final
+	 * values once the ENUMs are created.
+	 * 
+	 * @return
+	 */
+	public JPanel createAddMetricPanel() {
+		addNewMetricPanel.removeAll();
+		addNewMetricPanel.setLayout(new GridLayout(1, 6));
+		condition = new JComboBox<>();
+		setConditionVisibility();
+		JLabel ifCondition = new JLabel("IF", SwingConstants.CENTER);
+		JComboBox<String> value = new JComboBox<>();
+		value.addItem("LOC");
+		value.addItem("MET");
+		
+		JComboBox<String> comparison = new JComboBox<>();
+		comparison.addItem(">");
+		comparison.addItem("<");
+		comparison.addItem("=");
+		comparison.addItem("!=");
+		
+		JTextField threshold = new JTextField("10");
+		
+		JButton addMetricButton = new JButton("Add");
+		addMetricButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (value.getItemCount() == 0) {
+					JOptionPane.showMessageDialog(null, "Please check if all metric fields were entered correctly!");
+				} else {
+					String metric;
+					if (ruleMetrics.isEmpty()) {
+						metric = "IF " + value.getSelectedItem() + " " + comparison.getSelectedItem() + " "
+								+ threshold.getText();
+						
+					} else {
+						metric = condition.getSelectedItem() + " IF " + value.getSelectedItem() + " "
+								+ comparison.getSelectedItem() + " " + threshold.getText();
+					}
+					System.out.println("Add metric button pressed!");

```

Comment:
if these were only for debugging, please remove all prints

Reply:
They were indeed mostly for debugging. I kept them because I thought they might be useful later on, when adding functionality to this GUI, and since they're console prints, they wouldn't really be viewed by a regular user anyway. I'll delete for now though, if need be they're easy to add back in regardless.